tecton_client/_internal/tecton_client.py

```diff
@@ -0,0 +1,160 @@
+        request_data = self._build_get_features_request(
+            feature_service_id,
+            feature_service_name,
+            join_key_map,
+            request_context_map,
+            workspace_name,
+            metadata_options,
+            allow_partial_results,
+            request_options,
+        )
+        resp = self._client.post(url=urljoin(self._base_url, "feature-service/get-features"), json=request_data)
+
+        try:
+            resp.raise_for_status()
+        except HTTPStatusError as exc:
+            raise convert_exception(exc) from exc
+
+        return GetFeaturesResponse.from_dict(json.loads(resp.text))
+
+    def get_feature_service_metadata(
+        self,
+        *,
+        feature_service_name: Optional[str] = None,
+        feature_service_id: Optional[str] = None,
+        workspace_name: Optional[str] = None,
+    ) -> GetFeatureServiceMetadataResponse:
+        self._validate(feature_service_id, feature_service_name, workspace_name)
+        request_data = self._build_get_feature_service_metadata_request(
+            feature_service_id=feature_service_id,
+            feature_service_name=feature_service_name,
+            workspace_name=workspace_name,
+        )
+        resp = self._client.post(url=urljoin(self._base_url, "feature-service/metadata"), json=request_data)
+        try:
+            resp.raise_for_status()
+        except HTTPStatusError as exc:
+            raise convert_exception(exc) from exc
+
+        return GetFeatureServiceMetadataResponse.from_dict(json.loads(resp.text))
+
+    def _build_get_features_request(
+        self,
+        feature_service_id,
+        feature_service_name,
+        join_key_map,
+        request_context_map,
+        workspace_name,
+        metadata_options,
+        allow_partial_results,
+        request_options,
+    ):
+        workspace_name = workspace_name or self.default_workspace_name
+        params = {
+            "workspaceName": workspace_name,
+            "featureServiceName": feature_service_name,
+            "featureServiceId": feature_service_id,
+            "joinKeyMap": join_key_map or {},
+            "requestContextMap": request_context_map or {},
+            "allowPartialResults": allow_partial_results,
+        }
+        if metadata_options:
+            params["metadataOptions"] = {
+                # these two default to True
+                "includeNames": metadata_options.get(MetadataOptions.include_names, True),
+                "includeDataTypes": metadata_options.get(MetadataOptions.include_data_types, True),
+                # the rest default to False
+                "includeEffectiveTimes": metadata_options.get(MetadataOptions.include_effective_times, False),
+                "includeSloInfo": metadata_options.get(MetadataOptions.include_slo_info, False),
+                "includeServingStatus": metadata_options.get(MetadataOptions.include_serving_status, False),
+            }
```

Comment:
Maybe we can turn this logic into a `classmethod` of MetadataOptions? (and similar for RequestOptions)